octobot_trading/exchanges/util/parser/positions_parser.py

```diff
@@ -0,0 +1,356 @@
+            not_found_val=TraderPositionType.CROSS,
+        )
+
+    def _parse_original_side(self):
+        self._try_to_find_and_set(
+            PositioCols.ORIGINAL_SIDE.value,
+            [ExchangeCols.SIDE.value, BINANCE_POSITION_TYPE],
+        )
+
+    def _parse_side(self):
+        if (
+            self.formatted_record.get(PositioCols.POSITION_MODE.value)
+            is PositionMode.ONE_WAY
+        ):
+            # todo is it good to keep it like that in ONE_WAY mode?
+            self.formatted_record[PositioCols.SIDE.value] = PositionSide.BOTH
+        else:
+            self.formatted_record[PositioCols.SIDE.value] = (
+                PositionSide.LONG
+                if self.formatted_record.get(PositioCols.ORIGINAL_SIDE.value)
+                == PositionSide.LONG.value
+                else PositionSide.SHORT
+            )
+
+    def _parse_size(self):
+        def size_found(_size):
+            if (
+                self.formatted_record[PositioCols.SIDE.value] == PositionSide.LONG
+                or self.formatted_record[PositioCols.SIDE.value] == PositionSide.BOTH
+            ):
+                return _size
+            # short - so we set it to negative if it isn't already
+            return -_size if _size > 0 else _size
+
+        self._try_to_find_and_set_decimal(
+            PositioCols.SIZE.value,
+            [ExchangeCols.CONTRACTS.value, BINANCE_QUANTITY],
+            parse_method=size_found,
+            not_found_val=constants.ZERO,
+        )
+
+    def _parse_quantity(self):
+        self._try_to_find_and_set_decimal(
+            PositioCols.QUANTITY.value, [ExchangeCols.CONTRACT_SIZE.value]
+        )
+
+    def _parse_contract_type(self):
+        try:
+            self.formatted_record[
+                PositioCols.CONTRACT_TYPE.value
+            ] = self.exchange.get_contract_type(
+                self.formatted_record[PositioCols.SYMBOL.value]
+            )
+        except Exception as e:
+            self.exchange.logger.error(
+                f"Failed to parse position attribute: contract type with symbol: "
+                f"{self.formatted_record[PositioCols.SYMBOL.value]}"
+                f"with get_contract_type - error: {e}"
+            )
+
+    def _parse_timestamp(self):
+        def timestamp_not_found():
+            try:
+                return int(self.exchange.connector.get_exchange_current_time())
+            except Exception as e:
+                self.exchange.logger.error(
+                    f"Failed to parse position attribute: timestamp - "
+                    f"with get_exchange_current_time - error: {e}"
+                )
+
+        self._try_to_find_and_parse_with_method(
+            PositioCols.TIMESTAMP.value,
+            [ExchangeCols.TIMESTAMP.value],
+            int,
+            not_found_method=timestamp_not_found,
+        )
+
+
+# keep keys here from exchanges that are 100% safe
+BYBIT_BANKRUPTCY_PRICE = "bust_price"
```

Comment:
I don't think we should have any exchange specific code in OctoBot-Trading

Reply:
And what about adding synonyms for keys see latest commit in positions_parser on the bottom?
The idea is to only add synonyms that will 100% not cause issues

Reply:
And what about adding synonyms for keys see latest commit in positions_parser on the bottom?
The idea is to only add synonyms that will 100% not cause issues